backend/src/main/java/com/google/sps/datastore/UserDatastore.java

```diff
@@ -0,0 +1,48 @@
+package com.google.sps.datastore;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.data.User;
+
+/** Separates datastore method calls involving User type from caller */
+public final class UserDatastore {
+
+  // Datastore Key/Property constants
+  private static final String KIND_USER = "User";
+  private static final String PROPERTY_USERNAME = "username";
+  private static final String PROPERTY_TIMESTAMP = "timestamp";
+
+  /** Formatter for converting between String and ZonedDateTime */
+  private static final DateTimeFormatter formatter = DateTimeFormatter.ISO_ZONED_DATE_TIME;
+
+  /** Datastore */
+  private final DatastoreService datastore;
+
+  /** Constructor that takes in DatastoreService */
+  public UserDatastore(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  /** Put user in datastore. Overwrite user entity if user with same username already exists. */
+  public void addUser(User user) {
+    // Set properties of entity based on user fields
+    Entity userEntity = new Entity(KIND_USER, user.getUsername());
+    userEntity.setProperty(PROPERTY_USERNAME, user.getUsername());
+    userEntity.setProperty(PROPERTY_TIMESTAMP, user.getTimestamp());
+
+    // Insert entity into datastore
+    datastore.put(userEntity);
+  }
+
```

Comment:
you'll also want a method to get users from the database, likely keyed by usernsme

Reply:
Got it, added. I wasn't sure if I should add them yet since I don't know how exactly I'll be using them.